components/eventing-controller/pkg/handlers/jetstream.go

```diff
@@ -206,11 +202,23 @@ func (js *JetStream) SyncSubscription(subscription *eventingv1alpha1.Subscriptio
 				log.Debugw("skipping creating subscription on JetStream because it already exists", "subject", subject)
 				continue
 			}
-		}
-
-		// async callback for maxInflight messages
-		asyncCallback := func(m *nats.Msg) {
-			go callback(m)
+			log.Debugw("recreating subscription on JetStream because it was invalid", "subject", subject)
+			// bind the existing consumer to a new subscription on JetStream
+			jsSubscription, err := js.jsCtx.Subscribe(
+				jsSubject,
+				asyncCallback,
+				nats.Bind(js.config.JSStreamName, jsSubKey.ConsumerName()),
+			)
+			if err != nil {
+				if err != nats.ErrConsumerNotFound {
+					log.Errorw("failed to subscribe on JetStream", "subject", subject, "error", err)
+				}
+			} else {
```

Comment:
Here we log the error, how will this affect the overall Kyma subscription status? will it be ready if one of the corresponding JetStream subscriptions are not active?

Reply:
I think I should rewrite this error as: "failed to bind the consumer"
Basically this would not affect the JetStream subscription as it would create a new consumer in line 225.